optuna_dashboard/artifact/_backend.py

```diff
@@ -123,8 +127,17 @@ def delete_artifact(study_id: int, trial_id: int, artifact_id: str) -> dict[str,
             return {"reason": "Cannot access to the artifacts."}
         artifact_store.remove(artifact_id)
 
-        attr_key = _artifact_prefix(trial_id) + artifact_id
-        storage.set_study_system_attr(study_id, attr_key, json.dumps(None))
+        # The metadata of the artifact is stored in one of the following three locations:
+        storage.set_study_system_attr(
```

Comment:
[question] Should we check that `_artifact_prefix(trial_id) + artifact_id` exists in study_system_attr?

Reply:
I believe this is necessary to maintain backward compatibility since artifact meta information stored before this change are in  `study_system_attr`.

Reply:
Ok. I understand it.

It means that we will allow optuna-dashboard to be the following situations:

The behavior before this PR:
1. By using `upload_artifact_api`, key=`_artifact_prefix(trial_id) + artifact_id` in study_system_attr has some value.
2. By calling `delete_artifact`, key=`_artifact_prefix(trial_id) + artifact_id` in study_system_attr has `None`.

The behavior after this PR:
1. By using `upload_artifact_api`, key=`_artifact_prefix(trial_id) + artifact_id` in study_system_attr has some value with older version of optuna-dashboard, or  key=`ARTIFACTS_ATTR_PREFIX + artifact_id` in trial_system_attr has some value with newer version on optuna-dashboard.
2. By calling `delete_artifact`, both key=`_artifact_prefix(trial_id) + artifact_id` in study_system_attr and  key=`ARTIFACTS_ATTR_PREFIX + artifact_id` in trial_system_attr have  `None` even if they did not store any value in 1.

From my view, it is a little bit redundant that both study_system_attr and trial_system_attr have `None`, but it will work in the current implementation. I am ok with the current implementation.